fem/linearform_ext.cpp

```diff
@@ -51,11 +51,12 @@ void LinearFormExtension::Assemble()
       {
          // scan the attributes to set the markers to 0 or 1
          const int NE = fes.GetNE();
-         const auto attr = attributes.Read();
+         const auto attr = attributes->Read();
          const auto dimk = domain_integs_marker_k->Read();
          auto markers_w = markers.Write();
          mfem::forall(NE, [=] MFEM_HOST_DEVICE (int e)
          {
+            // TODO: should this also check for positive attributes?
```

Comment:
This is considered a user error -- if there are elements with attributes <= 0. We can check this outside the loop by looking at the minimal (the first) element attribute in `Mesh::attributes`. If we want to be even more direct, we can also check (in debug mode only, probably) if `attr[e] <= 0` and abort.